### PR TITLE
Add authentication to mutation

### PR DESCRIPTION
**Pull request by AroneyS**

### CODE REVIEW

**Comment by AroneyS** [comment]
I think graphQL isn't supposed to return HTTP error codes. Just the error message in the response

File: app/controllers/graphql_controller.rb [hunk 1]
```diff
@@ -28,14 +30,25 @@ def admin_status
...
   def setup_and_validate_credential
     authorization = request.headers['HTTP_AUTHORIZATION']
-    match = /\ABearer (.*)\Z/.match(authorization)
-
-    return unless match
-    auth = match[1]
-    temp = Credential.new(auth)
-    @credential = temp if temp.ok?
+    
+    if match = /\ABearer (.*)\Z/.match(authorization)
```
[the remark this answers is not in the document]
Comment (AroneyS) [reply app/controllers/graphql_controller.rb]: They both seem to work similar except if I have newline at the end of the string, `/Z` still matches it and `/z` doesn't. Not sure which is desired.

[the remark this answers is not in the document]
Comment (AroneyS) [reply app/controllers/graphql_controller.rb]: Can't think of any reason. So is `/z` the safer option then?

[the remark this answers is not in the document]
Comment (camallen) [reply app/controllers/graphql_controller.rb]: yep

**Pull request merged**